### PR TITLE
Fix bug in webhook handling found by KalleOlaviNiemitalo

### PR DESCRIPTION
**Pull request by dkjellin**
@KalleOlaviNiemitalo found  bug in how we updated webhooks. The test we had was a bit too simple. I've fixed this now and updated the test.

### CODE REVIEW

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/register/BitbucketWebhookHandler.java [hunk 1]
```diff
@@ -139,10 +139,10 @@ private BitbucketWebhook process(WebhookRegisterRequest request,
                                      Collection<BitbucketWebhookEvent> events) {
         String callback = constructCallbackUrl(request);
         List<BitbucketWebhook> webhooks = webhookClient.getWebhooks().collect(Collectors.toList());
-        Set<BitbucketWebhook> serverSideWebhooks =
+        List<BitbucketWebhook> serverSideWebhooks =
                 webhooks.stream()
                         .filter(hook -> hook.getName().equals(request.getName()) && hook.getUrl().equals(callback))
-                        .collect(Collectors.toSet());
+                        .collect(Collectors.toList());
```
Comment (dkjellin) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/register/BitbucketWebhookHandler.java:145]: This is built around the assumption that the `webhooks` will not contain duplicate webhooks. I think that is a safe assumption.

Comment (mhenschke-atl) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/register/BitbucketWebhookHandler.java:145]: You could alternatively set the parameter type on `update` to a Collection, and do a bit of stream wrangling to pull out the first element, but I think this is fine.

**Review by mhenschke-atl — APPROVED** [review]

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/register/BitbucketWebhookHandler.java (hunk 1 above)
Comment (dkjellin) [reply src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/register/BitbucketWebhookHandler.java:145]: _Probably_ I need it as a list for the update later on, so I figured there was no reason to not use a list.

**Review by khughes-atlassian — APPROVED** [review]

**Review by EliSalter-atl — COMMENTED** [review]
A post merge approval

**Pull request merged**